python/sparkdl/udf/keras_image_model.py

```diff
@@ -131,20 +134,17 @@ def _serialize_and_reload_with(preprocessor):
     """
     def udf_impl(spimg):
         import numpy as np
-        from PIL import Image
         from tempfile import NamedTemporaryFile
-        from sparkdl.image.imageIO import imageArrayToStruct, imageType
+        from sparkdl.image.imageIO import imageArrayToStruct
 
-        pil_mode = imageType(spimg).pilMode
-        img_shape = (spimg.width, spimg.height)
-        img = Image.frombytes(pil_mode, img_shape, bytes(spimg.data))
+        img = imageIO.imageStructToPIL(spimg)
         # Warning: must use lossless format to guarantee consistency
         temp_fp = NamedTemporaryFile(suffix='.png')
         img.save(temp_fp, 'PNG')
         img_arr_reloaded = preprocessor(temp_fp.name)
         assert isinstance(img_arr_reloaded, np.ndarray), \
             "expect preprocessor to return a numpy array"
-        img_arr_reloaded = img_arr_reloaded.astype(np.uint8)
+        img_arr_reloaded = img_arr_reloaded.astype(np.uint8)[...,::-1]
```

Comment:
i think we don't use ImageSchema in the context of the udfs, do we? i forget if the user ever gets access to the column with the image loaded throughout the udf using process. i'm wondering if we need to do this BGR - RGB back-and-forth here, especially given that a typical user will probably define a keras_load function using the RGB format.

Reply:
The udf returns image shcema so it has to be reversed.